### ISSUE TITLE
Pathfinder rewrite

### ISSUE THREAD

**Issue by metalgearsloth** [space-wizards/space-station-14#44445]
### Description

As far as I see it:
A) We need grouping / regions back in somehow. There's no way to do fast accessibility checks of entity A to entity B. Large empty rooms should be grouped together, maybe walls as well just to make edges easier.
B) Need some system to traverse map <> grid as they need separate pathfinders. Also need a map pathfinder.
C) Need tests, for now high-level ones are fine.
D) I forgot if pruning was still in but returning pruned or unpruned would be good
E) Remove async and use events.
F) Need to keep the debug experience good.
G) Partial paths; I should be able to get back a partial path that can be used before a full one is returned.
H) Same as current, parallel + time-sliced (might need cpujob changes to make it easier).
I) Portals same as current. navmesh portals not ingame portals so stuff like docks or ladders still work.

All of the above is separate to steering + thonk but the way I see it prio 1 for NPC work to clean it up.

### Reproduction

why is this mandatory dog bruh
